site/src/components/pages/dashboard/DashboardCard.tsx

```diff
@@ -1,31 +1,166 @@
+  colorGradient?: string;
   description: string;
   link: {
     title: string;
-    href: string;
+    href?: string;
+    onClick?: () => void;
   };
+  icon?: string;
+  variant?: "primary" | "secondary";
+};
+
+type CardWrapperProps = {
+  onClick?: () => void;
+  href?: string;
+};
+
+const CardWrapper: FC<CardWrapperProps> = ({ children, onClick, href }) => {
+  const sharedStyles: BoxProps["sx"] = useMemo(
+    () => ({
+      position: "relative",
+
+      "&::after": {
+        content: `""`,
+        position: "absolute",
+        top: 0,
+        left: 0,
+        height: "100%",
+        width: "100%",
+        boxShadow: 1,
+        opacity: 0,
+        transition: ({ transitions }) =>
+          transitions.create("opacity", { duration: 300 }),
+      },
+
+      "&:hover::after": {
+        opacity: 1,
+      },
+
+      "&:focus-visible": {
+        outline: ({ palette }) => `1px solid ${palette.purple[700]}`,
+      },
+    }),
+    [],
+  );
+
+  if (onClick) {
+    return (
+      <Box
+        component="button"
+        sx={{ textAlign: "left", ...sharedStyles }}
+        onClick={onClick}
+      >
+        {children}
+      </Box>
+    );
+  }
+
+  return (
+    <Link href={href} sx={sharedStyles}>
+      {children}
+    </Link>
+  );
+};
+
+export const DashboardCardSecondary: VoidFunctionComponent<
+  Pick<DashboardCardProps, "title" | "link" | "description" | "icon">
+> = ({ link, title, description, icon }) => {
+  return (
+    <CardWrapper href={link.href} onClick={link.onClick}>
+      <Box
+        sx={{
+          display: "flex",
+          border: ({ palette }) => `1px solid ${palette.gray[30]}`,
+          borderRadius: "4px",
+          p: 3,
+        }}
+      >
+        <Icon
+          sx={{ fontSize: 32, mr: 3, color: ({ palette }) => palette.gray[40] }}
+          className={icon}
+        />
+        <Box>
+          <Typography variant="bpLargeText" sx={{ mb: 0.75 }}>
+            {title}
+          </Typography>
+          <Typography sx={{ mb: 2 }}>{description}</Typography>
+          <Box
+            sx={{
+              color: ({ palette }) => palette.purple[700],
+              fontWeight: 600,
+              path: {
+                fill: "currentColor",
+              },
+              display: "flex",
+              alignItems: "center",
+            }}
+          >
+            <Box component="span" paddingRight={1}>
+              {link.title}
+            </Box>
+            <ArrowRightIcon
+              sx={{
+                width: "auto",
+                height: "0.8em",
+              }}
+            />
+          </Box>
+        </Box>
+      </Box>
+    </CardWrapper>
+  );
 };
 
 export const DashboardCard: VoidFunctionComponent<DashboardCardProps> = ({
   title,
   colorGradient,
   description,
   link,
+  variant,
+  icon,
 }) => {
+  if (variant === "secondary") {
+    return (
+      <DashboardCardSecondary
+        {...{ title, description, link, variant, icon }}
```

Comment:
This doesn't seem very readable. How about assigning `{ title, description, link, variant, icon }` to an object and spreading that object here?

Reply:
Thanks. Updated here 8423919